test/Container/ApplicationFactoryTest.php

```diff
@@ -1071,4 +1071,230 @@ public function testCanSpecifyRouteOptionsViaConfiguration()
         $this->assertInstanceOf(Route::class, $route);
         $this->assertEquals($expected, $route->getOptions());
     }
+
+    public function testExceptionIsRaisedInCaseOfInvalidRouteMethodsConfiguration()
+    {
+        $router = $this->prophesize('Zend\Expressive\Router\RouterInterface');
+        $emitter = $this->prophesize('Zend\Diactoros\Response\EmitterInterface');
+        $finalHandler = function ($req, $res, $err = null) {
+        };
+
+        $config = [
+            'routes' => [
+                [
+                    'path' => '/',
+                    'middleware' => 'HelloWorld',
+                    'allowed_methods' => 'invalid',
+                ],
+            ],
+        ];
+
+        $this->container
+            ->has('Zend\Expressive\Router\RouterInterface')
+            ->willReturn(true);
+        $this->container
+            ->get('Zend\Expressive\Router\RouterInterface')
+            ->will(function () use ($router) {
+                return $router->reveal();
+            });
```

Comment:
@nikolaposa Can you please update these to use `ZendTest\Expressive\ContainerTrait`, now that #194/#195 have been merged? Doing so will cut down on a lot of noise, making it easier to see what's being tested.

Thanks!